### ISSUE TITLE
Bug: "indent" does not work for comments in the line after function declaration

### ISSUE THREAD

**Issue by kkmuffme** [eslint/eslint#16952]
### Environment

Node version: 18.14.2
npm version: 9.5.1
Local ESLint version: 8.35.0
Global ESLint version: 8.35.0
Operating System: Centos 8

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration (as shown here https://eslint.org/docs/latest/rules/indent#options)</summary>

```
{
    "indent": ["error", "tab"]
}
```
</details>

```js
function foo() {
// comment
	console.log( 'hello' );
}
```

### What did you expect to happen?

Comment to be correctly indented

### What actually happened?

Comment wasn't indented

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Per our rule policies, stylistic rules are [frozen](https://github.com/eslint/eslint#stylistic-rule-updates) for new options unless an option is the only way to fix a bug or support a newly-added ECMAScript feature. You can always copy the core `indent` rule into a custom rule and make it behave the way you want or add new options to it.
